src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,688 +1,570 @@
+    2: Sort BAM
+    3: Index BAM
+
+    :param JobFunctionWrappingJob job: Toil Job instance
+    :param str uuid: Unique identifier for the sample
+    :param str url: URL to BAM file
+    :param Namespace config: Configuration options for pipeline
+    :param str rg_line: RG line for BWA alignment. Default is None.
+    :return: BAM and BAI FileStoreIDs
+    :rtype: tuple
+    """
+
+    cores = multiprocessing.cpu_count()
+
+    # Determine extension on input file
+    base, ext1 = os.path.splitext(url)
+    _, ext2 = os.path.splitext(base)
+
+    # Produce or download BAM file
+    if {ext1, ext2} & {'.fq', '.fastq'}:
+        if not config.run_bwa:
+            raise ValueError('Not configured to run BWA. Please set run-bwa: True')
+        elif not rg_line:
+            raise ValueError('FASTQ file requires an RG line for BWA alignment')
+        else:
+            get_bam = job.wrapJobFn(setup_and_run_bwa_kit, url, config).encapsulate()
+
+    elif {ext1, ext2} & {'.bam'}:
+        job.fileStore.logToMaster("Downloading BAM: %s" % uuid)
+        get_bam = job.wrapJobFn(download_url_job,
+                                url,
+                                name='toil.bam',
+                                s3_key_path=config.ssec,
+                                disk=config.file_size).encapsulate()
     else:
-        job.addFollowOnJobFn(index, ids, input_args)
-
-
-def index(job, shared_ids, input_args):
-    """
-    Index sample bam using samtools, calls haplotypeCaller.
-
-    :param job: Job instance
-    :param shared_ids: dictionary of shared file promises
-    :param input_args: dictionary of input arguments
-    """
-    work_dir = job.fileStore.getLocalTempDir()
-    # Retrieve file path
-    bam_path = return_input_paths(job, work_dir, shared_ids, 'toil.bam')
-    output_path = os.path.join(work_dir, 'toil.bam.bai')
-    # Call: index the normal.bam
-    parameters = ['index', 'toil.bam']
-    inputs=['toil.bam']
-    outputs={'toil.bam.bai': None}
-    docker_call(work_dir = work_dir,
-                parameters = parameters,
-                tool = 'quay.io/ucsc_cgl/samtools',
-                inputs=inputs,
-                outputs=outputs)
-    # Update FileStore and call child
-    shared_ids['toil.bam.bai'] = job.fileStore.writeGlobalFile(output_path)
-    job.addChildJobFn(haplotype_caller, shared_ids, input_args, cores = input_args.cpu_count)
-
-
-def haplotype_caller(job, shared_ids, input_args):
+        raise ValueError('Accepted formats: FASTQ or BAM\n'
+                         'Could not determine {} file type from URL:\n{}'.format(uuid, url))
+
+    rm_secondary = job.wrapJobFn(run_samtools_view,
+                                 get_bam.rv(),
+                                 flag='0x800',
+                                 ncores=cores,
+                                 disk=config.file_size, cores=cores)
+
+    sort_bam_disk = PromisedRequirement(lambda x: 3*x.size + human2bytes('10G'), rm_secondary.rv())
```

Comment:
Where ever a constant value (e.g., the '8G' in this line) comes in, you need to document where this constant comes from.